exercises/grade-school/test/test_grade_school.c

```diff
@@ -61,9 +52,25 @@ static void test_adding_student_adds_to_roster(void)
    check_roster(&input, &expected);
 }
 
-static void test_adding_more_students_adds_to_roster(void)
+static void test_a_student_cant_be_in_two_different_grades(void)
 {
-   TEST_IGNORE();
+   roster_t input = {
+      2, {
+          (student_t) {2, "Aimee"},
+          (student_t) {1, "Aimee"}}
+   };
+   uint8_t desired_grade = 5;
+
+   populate_roster(&input);
+
+   roster_t actual = get_grade(desired_grade);
+
+   TEST_ASSERT_EQUAL(0, actual.count);
+}
```

Comment:
This test is really confusing. How does this ensure that a student can't be in two different grades? `"Aimee"` is added to grades 1 and 2, but then it checks that grade 5 is empty.

Reply:
Whoops, copy-pasta error

Reply:
Looking at this some more, the way this test is specified makes me think I should rearchitect this exercise completely.

My first inclination here was to check student names as a student is added, compare it to names already on the list and not add it if the name already exists.

However this supposes a linear timeline of added the input (i.e. add Aimee at grade 2 that succeeds, then attempt to add Aimee at grade 1 that fails). However, the desired grade is specified as 2 with an expected output of zero (implying that actually the addition of Aimee in grade 2 should have failed).

This suggests a much simpler interface like the following, where there is no local store of the roster, it is instead supplied as an input each time a query is made.

```c
typedef struct {
   uint8_t grade;
   char *name;
} student_t;

typedef struct {
   size_t count;
   student_t students[MAX_STUDENTS];
} roster_t;

roster_t get_roster(roster_r students);

roster_t get_grade(uint8_t grade, roster_r students);
```